pkg/sensors/exec/procevents/proc_reader.go

```diff
@@ -408,6 +463,20 @@ func listRunningProcs(procPath string) ([]procs, error) {
 			if err != nil {
 				logger.GetLogger().WithError(err).Warnf("Reading Loginuid of %s failed, falling back to loginuid: %d", pathName, uint32(auid))
 			}
+
+			// Check and add only thread leader processes.
+			if len(status.NSpid) > 0 && len(status.NStgid) > 0 {
```

Comment:
this is unrelated change, right? I think it should be in separated commit or at least mentioned in the changelog

Reply:
Yes, I will do that in a separate commit.

Reply:
I didn't get this change! IIRC scanning /proc/ only shows thread leaders, it should not show threads or maybe I'm getting old and making stuff...

Reply:
I believe that you are correct. I have checked the logs from running this several days with many restarts and no similar log messages were found. 

I initially did that as a sanity check but it seems that we can remove that. Thanks!